pkg/templates/generic_lookup_func.go

```diff
@@ -297,3 +299,103 @@ func onAllowlist(allowlist []ClusterScopedObjectIdentifier, rsrc ClusterScopedOb
 
 	return false
 }
+
+func (t *TemplateResolver) getNodesWithExactRolesHelper(
+	options *ResolveOptions,
+	templateResult *TemplateResult,
+) func(...string) (
+	map[string]interface{}, error,
+) {
+	return func(name ...string) (
+		map[string]interface{}, error,
+	) {
+		return t.getNodesWithExactRoles(options, templateResult, name...)
+	}
+}
+
+// function getNodesWithExactRoles gets a listing of all nodes with only the
+// role specified.  Any nodes that include other roles in addition
+// to the specified roles are not included.
+func (t *TemplateResolver) getNodesWithExactRoles(
+	options *ResolveOptions,
+	templateResult *TemplateResult,
+	name ...string,
+) (
+	map[string]interface{}, error,
+) {
+	result := []unstructured.Unstructured{}
+
+	if len(name) == 0 || strings.TrimSpace(name[0]) == "" {
+		return nil, fmt.Errorf("%w: name must be specified", ErrInvalidInput)
+	}
+
+	var searchRoles []string
+
+	for _, n := range name {
+		searchRoles = append(searchRoles, fmt.Sprintf("node-role.kubernetes.io/%v", n))
+	}
+
+	nodes, err := t.getOrList(options, templateResult, "v1", "Node", "", "", searchRoles...)
+	if err != nil {
+		return nil, err
+	}
+
+	var nodeList unstructured.UnstructuredList
+
+	nlByte, err := json.Marshal(nodes)
+	if err != nil {
+		return nil, err
+	}
+
+	_ = json.Unmarshal(nlByte, &nodeList)
```

Comment:
Does this work instead? I think it'd be more efficient.
```suggestion
	nodeList := unstructured.UnstructuredList{}

	nodeList.SetUnstructuredContent(nodes)
```